### ISSUE TITLE
feature request for codecov.yaml: ability to disable codecov from commenting on PRs made by a specific author

### ISSUE THREAD

**Issue by vaneyckt** [codecov/support#118]
Hey all,

at work we have certain automation accounts that help us keeping everyone's pull requests in sync. For example, whenever someone pushes code to the `master` branch, we have a tool that will open pull requests from `master` against all branches that are actively being worked on.

Unfortunately all these pull requests will have the same last commit at the head branch of the PR (that is, the last commit of the master branch). Because of the way Github deals with commit statusses, setting the commit status of the last commit of one of these PRs, will cause Github to set that commit status for all thse PRs. This in turn will cause CodeCov to assign the code coverage information of this one PR, to all these PRs. It's very hard for CodeCov to deal with this issue, as Github internally does not associate commit status information with a Commit of a Pull Request. Instead it just associates it with a Commit. 

Long story short, would it be possible to tell CodeCov to not comment on PRs made by certain accounts, so as to prevent me from having to explain to people over and over again about why the wrong coverage information gets associated with Pull Requests made by some of our automation accounts?

**Comment by stevepeak** [comment]
> #105 

Is the only thing you want skipped the PR comment? Or should the entire commit be skipped. Because there are notifications such as Slack/Gitter/Email/etc. and these PR's will be reflected in the UI.

### Idea A
> Skipping the entire commit. Data is still stored, but it will not show up in Codecov UI nor post any features/notifications
```yml
advanced:
  skip_commits_from_authors:
    - @author_username
```

### Idea B
> Skip features

```
comment:
  ignore:
    - @author_usename
```

**Comment by vaneyckt** [comment]
Hi Steve,

after thinking about this for a while, it turns out we can solve this problem on our end. Would you mind if I closed this? I'm going to try and explain what happened here. This is gonna be a bit long and detailed, so please let me know if there is anything I left unclear :).

#### Introduction to mergebot
Our workflow at the office is as follows:
  - we have a `master` branch
  - whenever someone writes a new feature they branch of master and create a feature branch
  - however, not all features branches are branched from master. Some are branched from another feature branch!

Whenever a feature branch is finished, it gets merged into master. However, we want to make sure that the new code in master automagically makes it way into all feature branches. For this, we use an in-house tool called mergebot.

Mergebot allows developers to specify a mergebot flow in the form of a .yaml file. Example of a mergebot flow
```
- master
  - feature-A
  - feature-B

- feature-A
  - feature C
```
The above flow will cause mergebot (our Github automation account) to open pull requests from `master into feature-A` and from `master into feature-B` whenever new code gets merged into `master`. Similarly, it will open a pull request from `feature-A into feature-C` whenever new code gets merged into `feature-A`.

#### A closer look at what happens with these pull requests
Let's say our mergebot flow looks like this
```
- master
  - feature-A
  - feature-B
```
and let's say some code just got merged into master. This will cause mergebot to open two pull requests in github. Okay. As you know, we have tooling (Jently) in place that tests these pull requests and sets the commit  status in order to let us know whether these pull requests are safe to merge. Let's have a closer look at how this is done.

We have a pull request from `master into feature-A`. We tell our tooling that we want to test this pull request on Jenkins. So we create a throwaway merge commit that merges the `master` branch into the `feature-A` branch, and we tell Jenkins to test the code of this merge commit. Then, when all tests pass, we want to set the commit status of the pull request.

However, github does not provide any funtionality to set the commit status of a pull request. It provides [functionality to set the commit status of a commit](https://developer.github.com/v3/repos/statuses/), with the understanding that setting the commit status of the last commit of the head branch of a pull request will cause github to present that in the UI as the status of the pull request. This is horrible design by Github that causes bad things to happen!

#### Why is this horrible design?
Let's look at our pull requests again, but now let's look at the commits. If we have two pull requests, one of which wants to merge `master into feature-A` and another one which wants to merge `master into feature-B`, then both pull requests will have `master` as their head branch. Meaning that the last commit of the head branch of each pull request is just the last commit of master. They are identical!

So let's say we are testing the throwaway merge commit of the first of these two pull requests on Jenkins.
Then, when all tests pass, we set the commit status of the last commit of its head branch. Doing this causes Github to modify the commit status of BOTH pull requests, even though we have only tested the one pull request.

CodeCov gets hit with a similar problem as a consequence of the badly designed github commit status api. So in order to tell CodeCov to check the commit status for a given commit, I need to pass the relevant commit sha as a param. However, this commit sha will be the last commit of the head branch of a pull request. However, the actual coverage data generated by Jenkins was done on a throwaway merge commit. So now we have a scenario where CodeCov is associating coverage data (that was done on a merge commit) with the last commit of the head branch of a pull request (that is not guaranteed to be unique, as we can have multiple pull requests with the same last commit at their respective head branches).

#### How I have modified mergebot to prevent this
Let's say you have two open pull requests, graphically represented as follows
```
feature-A <- master   # merge master into feature-A
feature-B <- master   # merge master into feature-B
```
Obviously these pull requests have the problem Id escribed above, in that both of them will have the same last commit of the head branch - that is to say, the last commit of the `master` branch.

Instead I have modified our tools to create pull requests like so
```
feature-A <- (master <- feature-A)
feature-B <- (master <- feature-B)
```
That is to say, mergebot now creates a throwaway branch where we merge `feature-A` into `master`. We then open a pull request to merge this throwaway branch into `feature-A`. We do a similar process for the `feautre-B` pull request.

This way we can now be sure that all (mergeable) pull requests created by mergebot will never share the last commit of their head branch with another pull request.
